feat: update form inputs to match UI Kit v1.4.3

saad-mo opened the pull request:
Closes sap/fundamental#1498

Update input fields based on UI Kit

before: 
![Screen Shot 2019-06-19 at 4 35 36 PM](https://user-images.githubusercontent.com/15215400/59803232-84b71080-92b0-11e9-8c8c-e91b1996ddec.png)


after: 
![Screen Shot 2019-06-19 at 4 35 54 PM](https://user-images.githubusercontent.com/15215400/59803243-8b458800-92b0-11e9-8c0b-22cb924c2ceb.png)


#### Test

* http://localhost:3030/form
* http://127.0.0.1:4000/components/form.html

#### Changelog

**New**

* `--information` modifier for form messages 
* `--no-icon` modifier to render form messages without an icon

**Changed**

* Update input focus state
* Updated input Disabled state
* Additional support for error, warning, success, Information states of fields
* Italic text for placeholder text
* Upload File button for forms (new component, will be addressed in a separate ticket)
* Required label has a red *
* Input message now has background colors 
* Updated disabled state colors 
* Updated read-only state colors 

**Removed**

nothing

saad-mo commented:
@LeoT7508 

https://deploy-preview-1499--fundamental.netlify.com/components/form.html

droshev commented:
@saad-mo can you cherry pick this in fundamental-styles too(once it is approved)?

saad-mo commented:
@LeoT7508 

> Are the fields the correct height? 36px tall? Or is the height difference because of the highlighted outline?
yes default field height is 36px; 

> The message strip height should be 28px tall.
fixed

> The inside left padding should be 8px.
yes, this is correct 

> The Message text should be Text 1 color.
fixed

---

rest of the color updates will come in a separate PR
